### PR TITLE
Fixed a bug causing undeleted board to not have original title

### LINKED ISSUE #3327: Bug: Name of a deleted Board is not restored after using CMD/CTRL+Z to restore
Reproduced in Rainforest, server v7.1-rc2, Boards `main/master` branch

## Steps to reproduce the behavior

1. Open any board
2. Delete the board using the 3-dots to the right of the board name
3. Restore the board using CMD/CTRL+Z
4. Observed: Board and cards are restored but the board name is not restored. Instead, the board name is "Untitled Board"

## Expected behavior

Board and cards are restored and the board name is as per original board name

## Edition and Platform

 - Edition: Mattermost Boards (plugin)
 - Version: master/main
 - Browser and OS: macOS/Safari;  Windows 10/Chrome

## Additional context (optional)
* Sev 2: Affects major functionality with a difficult or non-obvious workaround

### PR DESCRIPTION
**Pull request by harshilsharma63**
Fixes #3327

### CODE REVIEW

File: server/services/store/storetests/boards.go [hunk 1]
```diff
@@ -803,9 +803,23 @@ func testUndeleteBoard(t *testing.T, store store.Store) {
...
 		board := &model.Board{
-			ID:     boardID,
-			TeamID: testTeamID,
-			Type:   model.BoardTypeOpen,
+			ID:              boardID,
+			TeamID:          testTeamID,
+			Type:            model.BoardTypeOpen,
+			Title:           "Dunder Mifflin Scranton",
+			MinimumRole:     model.BoardRoleCommenter,
+			Description:     "Bears, beets, Battlestar Gallectica",
```
Comment (jespino) [inline server/services/store/storetests/boards.go:811]: :heart:

File: server/services/store/sqlstore/board.go [hunk 2]
```diff
@@ -774,8 +774,8 @@ func (s *SQLStore) undeleteBoard(db sq.BaseRunner, boardID string, modifiedBy st
 		board.CreatedBy,
 		modifiedBy,
 		board.Type,
-		board.MinimumRole,
 		board.Title,
+		board.MinimumRole,
```
Comment (jespino) [inline server/services/store/sqlstore/board.go:778]: :facepalm:

**Review by jespino — APPROVED** [review]
LGTM

File: server/services/store/sqlstore/board.go (hunk 2 above)
Comment (harshilsharma63) [reply server/services/store/sqlstore/board.go:778]: 😄

**Pull request merged**